endstate_correction/simulation/base.py

```diff
@@ -0,0 +1,139 @@
+        ml_atoms: List[int],
+        protocol: BSSProtocol,
+        name: str = "endstate_correction",
+        work_dir: str = "./",
+        potential: str = "ani2x",
+        interpolate: bool = True,
+    ):
+        self.logger = logging.getLogger("EndstateCorrectionBase")
+        self.top = top
+        self.protocol = protocol
+        self.env = env
+        self.name = name
+        self.work_dir = work_dir
+        self._mm_topology = None
+        self._initial_coordinates = None
+
+        mm_system = self._createSystem()
+        ml_potential = MLPotential(potential)
+        ml_system = ml_potential.createMixedSystem(
+            self.get_mm_topology().topology,
+            mm_system,
+            ml_atoms,
+            interpolate=interpolate,
+            implementation="nnpops",
+        )
+        integrator = self.get_integrator()
+        _, platform = check_implementation()
+        platform = Platform.getPlatformByName(platform)
+        self.simulation = Simulation(
+            self.get_mm_topology().topology, ml_system, integrator, platform=platform
+        )
+
+    def get_mm_topology(self) -> Topology:
+        if self._mm_topology is None:
+            self._mm_topology = self._get_mm_topology()
+        return self._mm_topology
+
+    def get_initial_coordinates(self) -> unit.Quantity:
+        if self._initial_coordinates is None:
+            self._initial_coordinates = self._get_initial_coordinates()
+        return self._initial_coordinates
+
+    @abc.abstractmethod
+    def _get_mm_topology(self) -> Union[CharmmPsfFile, AmberPrmtopFile]:
+        pass
+
+    @abc.abstractmethod
+    def _get_initial_coordinates(self) -> unit.Quantity:
+        pass
+
+    def get_integrator(self) -> Integrator:
+        return LangevinIntegrator(
+            self.protocol.temperature,
+            self.protocol.collision_rate,
+            self.protocol.timestep,
+        )
+
+    def _createSystem(self, **kwargs):
+        if self.env == "vacuum":
+            mm_system = self.get_mm_topology().createSystem(
+                nonbondedMethod=NoCutoff, **kwargs
+            )
+        else:
+            mm_system = self.get_mm_topology().createSystem(
+                nonbondedMethod=PME,
+                nonbondedCutoff=self.protocol.rlist * unit.nanometers,
+                **kwargs,
+            )
+        return mm_system
+
+    def start(self):
+
+        # path where samples should be stored (will be created if it doesn't exist)
+        base = f"{self.work_dir}/equilibrium_samples/"
+        os.makedirs(base, exist_ok=True)
+        # define lambda states
+        lamb = self.protocol.lam["ml-lambda"]
+        self.logger.info(f"{lamb=}")
+        # define where to store samples
+        trajectory_file = f"{base}/{self.name}_samples_{self.protocol.n_integration_steps}_lamb_{lamb:.4f}_{self.env}.dcd"
+        self._traj_file = trajectory_file
+        self.logger.info(f"Trajectory saved to: {trajectory_file}")
```

Comment:
changed the name of the trajectory

Reply:
So currently in our RBFE floe, the trajectory is saved using the naming convention of `f"{ligand_name}/{stage}/lambda_{lambda_index}/amber.nc"` where `stage` could be `Minimisation`, `Equilibration` or `Production`. 

It would be nice if the interface allowed me to keep the same naming convention so the data collecting could be easy. So my thought is that EndstateCorrectionBase(work_dir=f"{ligand_name}/{stage}", name="openMM"), which would store the trajectory in `f"{ligand_name}/{stage}/lambda_{lambda_index}/openMM.dcd"`.

If the {self.protocol.n_integration_steps} is needed, do you think if this could be done via the `work_dir`? Like work_dir=f"{self.protocol.n_integration_steps}" then it would allow you to keep this information. One reason that I don't like variable file name is that it would allow later stage analysis to be difficult as one might not necessarily know what would the file name be without having access to information like n_integration_steps.

Reply:
I see! I wasn't aware of that! In that case there is a good reason for a renaming and I appreciate the explanation.
I think the naming convention `f"{ligand_name}/{stage}/lambda_{lambda_index}/openMM.dcd"` is fine, we can skip the `n_integration_steps` in that case. Also, if you prefer, we can switch from `dcd` to `nc`? Or at least provide the option to do so?